### PR TITLE
IB/AMO: removed duplicating indirect AMO conf

### PR DESCRIPTION
**Pull request by hoopoepg**
- removed duplicating indirect atomics/UMR configuration

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7967/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_md.c [hunk 1]
```diff
@@ -885,7 +881,7 @@ static ucs_status_t uct_ib_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
     if ((memh->flags & UCT_IB_MEM_ACCESS_REMOTE_ATOMIC) &&
         !(memh->flags & UCT_IB_MEM_FLAG_ATOMIC_MR) &&
         (memh != &md->global_odp) &&
-        md->config.enable_umr_pack)
+        md->config.enable_indirect_atomic)
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c]: is there another place where enable_indirect_atomic is used? maybe this check here should be removed at all?

Comment (hoopoepg) [reply src/uct/ib/base/ib_md.c]: it should be enabled by default... how can we remove this check?

Comment (yosefe) [reply src/uct/ib/base/ib_md.c]: maybe https://github.com/openucx/ucx/blob/master/src/uct/ib/base/ib_md.c#L196 is sufficient?

Comment (hoopoepg) [reply src/uct/ib/base/ib_md.c]: then how we cover case when user force disable indirect atomics (even it is supported)?

Comment (yosefe) [reply src/uct/ib/base/ib_md.c]: looks to me the according to that code in umr_qp_create, if md->config.enable_indirect_atomic==0 it behaves as if it's unsupported

Comment (hoopoepg) [reply src/uct/ib/base/ib_md.c]: done

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7972/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7981/ for details (Mellanox internal link).

**Pull request merged**